modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -619,12 +616,26 @@ protected Assignment getAssignment(User.ID userID, Application.Name applicationN
                     Assignment.Status.EXPERIMENT_NOT_STARTED);
         }
 
+        Assignment assignment = getAssignment(experimentID, userID, context, userAssignments, bucketList);
+
+        // Case: Experiment expires -> existing Assignments are returned and new Assignments
+        // are returned as null assignments
         if (currentTime > experiment.getEndTime().getTime()) {
-            return nullAssignment(userID, applicationName, experimentID,
-                    Assignment.Status.EXPERIMENT_EXPIRED);
+
+            // the experiment moves to the PAUSED state if it was RUNNING before
+            if (Experiment.State.RUNNING == experiment.getState()) {
+                experimentUtil.updateExperimentState(experiment, Experiment.State.PAUSED);
+                metadataCache.refresh();
```

Comment:
@AndreaSuckro 👍  for refreshing the cache here - this cache had to be refreshed when in-memory state for an experiment is RUNNING but experiment is expired.